DiscordChatExporter.Domain/Discord/Models/Channel.cs

```diff
@@ -21,7 +22,7 @@ public enum ChannelType
     }
 
     // https://discord.com/developers/docs/resources/channel#channel-object
-    public partial class Channel : IHasId
+    public partial class Channel : IHasId, IComparable, IComparable<Channel>
```

Comment:
I pulled it out, but I couldn't figure out how to apply it to the XAML, which kept yelling at me that Channel didn't implement IComparable. The extent of my knowledge regarding WPF is everything I googled in the past 30 minutes, so in the end I just modified one of the sorts to use the Position property instead of Category, and that appears to be correctly sorting everything according to the Discord hierarchy. I don't know if the comparer class is still needed (or even if what I did is correct), so I'll await your input on that.

Reply:
Yeah, don't worry about that, I will fix it if it's broken later. It's not the main goal of the PR.